src/useScrollSync.ts

```diff
@@ -0,0 +1,91 @@
+import type React from "react";
+import { useEffect } from "react";
+
+interface Options {
+  proportional?: boolean;
+  axis?: "horizontal" | "vertical" | "both";
+}
+
+export const useScrollSync = (
+  refs: React.MutableRefObject<HTMLElement | null>[],
+  { proportional = true, axis = "both" }: Options = {}
+): void => {
+  useEffect(() => {
+    const nodes = refs
+      .map((ref) => ref.current)
+      .filter((n) => n !== null) as HTMLElement[];
+    nodes.map((node) => {
+      node.onscroll = () =>
+        handleScroll(
+          node,
+          nodes.filter((n) => n !== node),
+          { proportional, axis }
+        );
+    });
+    return () => {
+      nodes.map((node) => {
+        node.onscroll = null;
+      });
+    };
+  }, []);
```

Comment:
Missing dependencies here. When refs will change this effect won’t run.

Reply:
done